doc/debugging.rst

```diff
@@ -14,33 +14,91 @@ This allows breakpoints and the execution flow to be viewed and controlled right
+SCREENSHOT: (timeit)
+
+SCREENSHOT of highlighting step (close up)
+
+Finally, you can enter and execute multiline statements in Spyder's debugger just like with the regular IPython prompt, to easily execute complex code while debugging.
+
+GIF of multiline debugging
+
+
+
+===========
+Breakpoints
+===========
+
+Spyder's debugger is integrated with the :guilabel:`Breakpoints` pane, which lists the file, line, and condition (if any) of every breakpoint defined.
+To open it, select :menuselection:`Debug --> List breakpoints`, or press :kbd:`Ctrl-Shift-B` by default.
 
 .. image:: images/debugging/breakpoints-standard.png
    :alt: Spyder's Breakpoints panel, with a number of examples showing file, line number and an optional condition
 
-* A :guilabel:`Breakpoints` pane, listing the file, line, and condition (if any) of every breakpoint defined (:menuselection:`&Debug --> List breakpoints`, or :kbd:`Ctrl-Shift-B` by default).
+There are several different ways to set and clear breakpoints:
+
+  * With the :guilabel:`Set/clear breakpoint` option in the Debug menu.
+  * Through pressing the configurable keyboard shortcut (:kbd:`F12` for normal, or :kbd:`Shift-F12` for conditional breakpoints by default).
+  * By clicking to the left of the line number in an open file in the Editor (adding :kbd:`Shift` for a conditional breakpoint).
+  * With the ``breakpoint()`` function in your code.
+  * Interactively, using the ``!b`` command in an ``ipdb`` session.
+
+GIF: setting conditional breakpoint shift-clicking
 
-* Full GUI control over debugger execution from the :guilabel:`Debug` menu, :guilabel:`Debug toolbar` and via configurable keyboard shortcuts, along with the standard ``ipdb`` console commands.
+You can access and edit local and global variables at each breakpoint through the :doc:`variableexplorer`.
 
-.. image:: images/debugging/debugging-condbreakpoint.png
-   :alt: Inset of Spyder's Editor, with a breakpoint set and the condition dialog open
+SCREENSHOT with console and variable explorer showing local and global variables
 
-* Highlighting of the current frame (debugging step) in the :doc:`editor`.
 
-* The ability to access and edit local and global variables at each breakpoint through the :doc:`variableexplorer`, and run many commands in the :doc:`ipythonconsole`.
 
-For a comprehensive but accessible introduction to ``pdb``/``ipdb``, consult Steve Ferg's excellent online guide, `Debugging in Python`_.
+=================
+Advanced features
+=================
+
+You can avoid stepping through other Python packages while debugging by enabling the new :guilabel:`Ignore Python libraries while debugging` option in Spyder's preferences, under :menuselection:`IPython Console --> Debugger --> Debug`.
+This will skip all the built-in and third-party Python modules.
+
+SCREENSHOT with option in preferences
+
+If your code has variables with the same names as Pdb commands (e.g. ``a``, ``step``), you can still refer to them as normal while debugging.
```

Comment:
```suggestion
If your code has variables with the same names as Pdb commands (e.g. ``b`` or ``step``), you can still refer to them as normal while debugging.
```

* We agreed on using `b` for the screenshot below.
* `or` is missing to me.

Reply:
The or isn't necessary in this context (ordered list of examples), but its not wrong to have it with it